CHANGELOG.md

```diff
@@ -3,7 +3,61 @@
+- The bundle manifest can now include an extra `metadata` key where arbitrary key-value pairs can be stored. Authored by @[viovanov](https://github.com/viovanov)
+- The bundle plugin now supports file:// urls in the `resource` field for test purposes.
+- The decision log plugin emits a clearer message at DEBUG instead of INFO when there is no work to do. Authored by [andrewbanchich](https://github.com/andrewbanchich)
+- The discovery plugin now supports a `resource` configuration field like the bundle plugin. Similarly, the `resource` is treated as the canonical setting to identify the discovery bundle.
+
+### Tooling
+
+- The `opa test` timeouth as been increased to 30 seconds when benchmarking ([#3107](https://github.com/open-policy-agent/opa/issues/3107))
+- The `opa eval --schema` flag has been fixed to correctly set the schema when a _single_ schema file is passed
+- The `opa build --debug` flag output has been improved for readability
+- The `array.items` JSON schema value is now supported by the type checker
+- The `opa fmt` subcommand can now exit with a non-zero status when a diff is detected
+- The `opa test` subcommand no longer emits bogus file paths when fed a file:// url
+
+### Built-in Functions
+
+- The `http.send` built-in function falls back to the system certificate pool when the `tls_ca_cert` or `tls_ca_cert_env_variable` options are not specified ([#2271](https://github.com/open-policy-agent/opa/issues/2271)) authored by @[olamiko](https://github.com/olamiko)
+
+### Evaluation
+
+- The order of support rules emitted by partial evaluation is now deterministic ([#3453](https://github.com/open-policy-agent/opa/issues/3453)) authored by @[andrehaland](https://github.com/andrehaland)
+- The big number performance regression caught by the fuzzer has been resolved ([#3262](https://github.com/open-policy-agent/opa/issues/3262))
+- The evaluator has been updated to memoize calls to rules with arguments (functions) within a single query. This avoids recomputing function results when the same input is passed multiple times (similar to how complete rules are memoized.)
+
+### WebAssembly
+
+- The `wasm` target no longer panics if the OPA binary does not include a wasm runtime ([#3264](https://github.com/open-policy-agent/opa/issues/3264))
+- The interrupt handling mechanism has been rewritten to make safe use of the wasmtime package. The SDK also returns structured errors now that are more aligned with topdown. ([#3225](https://github.com/open-policy-agent/opa/issues/3225))
+- The SDK provides the subset of required imports now (which is useful for debugging with opa_println in the runtime library if needed.)
+- The opa_number_float type has been removed from the value library (it was unused after moving to libmpdec)
+- The runtime library is now built using llvm-12 to resolve build issues with C++ dependencies
```

Comment:
```suggestion
- The runtime library is now built using llvm-12
```

The resolved C++ issues were a result of bumping the LLVM version. The reason for doing _that_ was because it's nice to stay current, especially when it comes to Wasm.